### PR TITLE
ZPS: Add missing offsets

### PR DESCRIPTION
**Pull request by xerox8521**
Adds support for the SDKHook Types: OnTakeDamage_Alive, GroundEntChanged, GetMaxHealth, Blocked and Reload.
Adds support for GivePlayerAmmo native.

I would have added them earlier but wasn't aware they were missing entirely.

### CODE REVIEW

**Comment by xerox8521** [comment]
The last 2 commits weren't supposed to be in this PR. I wanted them in a seperate one and I'm not entirely sure how to split these.

File: extensions/sdktools/vnatives.cpp [hunk 1]
```diff
@@ -516,12 +516,24 @@ static cell_t ForcePlayerSuicide(IPluginContext *pContext, const cell_t *params)
 			return pContext->ThrowNativeError("\"CommitSuicide\" wrapper failed to initialize");
 		}
 	}
-
-	START_CALL();
-	DECODE_VALVE_PARAM(1, thisinfo, 0);
-	*(bool *)(vptr + pCall->vparams[0].offset) = false;
-	*(bool *)(vptr + pCall->vparams[1].offset) = false;
-	FINISH_CALL_SIMPLE(NULL);
+#if SOURCE_ENGINE == SE_SDK2013
```
Comment (KyleSanderson) [inline extensions/sdktools/vnatives.cpp:519]: Something is wrong here.

**Review by KyleSanderson — CHANGES_REQUESTED** [review]

File: extensions/sdktools/vnatives.cpp (hunk 1 above)
Comment (xerox8521) [reply extensions/sdktools/vnatives.cpp:519]: Can you clarify? Apart that the change was supposed to be in a different PR.
The check is basically like this one here: https://github.com/alliedmodders/sourcemod/blob/master/extensions/sdktools/vnatives.cpp#L405

Comment (psychonic) [reply extensions/sdktools/vnatives.cpp:519]: What is the issue with leaving bForce as false? Does it just not do anything?

Comment (peace-maker) [reply extensions/sdktools/vnatives.cpp:519]: The preprocessor directive you added makes the native do nothing on any other engine than sdk2013.
You could refactor it to use a `bool bForce` variable which defaults to `false` and is set to true for zps.

Comment (xerox8521) [reply extensions/sdktools/vnatives.cpp:519]: > What is the issue with leaving bForce as false? Does it just not do anything?

If set to false the action is delayed. So if you slay someone or timebomb explode the explosion happens and then nothing until the delay is finished.

> The preprocessor directive you added makes the native do nothing on any other engine than sdk2013. You could refactor it to use a `bool bForce` variable which defaults to `false` and is set to true for zps.

Oh yea I see will fix that up

**Comment by Headline** [comment]
@KyleSanderson updates?

**Review by psychonic — APPROVED** [review]

**Pull request merged**